app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -16,6 +16,16 @@ def create_resource(_type, _id = nil, data = {})
       action_result(false, err.to_s)
     end
 
+    def edit_resource(type, id, data = {})
+      raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
+      cloud_volume = resource_search(id, type, collection_class(:cloud_volumes))
+  
```

Comment:
We should have a supports_feature check that this cloud_volume supports being updated since out of the 7 providers which collect cloud_volumes only 3 support updating them.

It looks like the old UI probably used `CloudVolume#validate_update_volume` (and maybe [[still]](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/cloud_volume_controller.rb#L293) [[does]](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/cloud_volume_controller.rb#L310))

Reply:
@agrare Could you share a sample from api code where we are using similar check in any of the api calls, im not really sure how to do it, thanks.

Reply:
Of course, just look for anything that does `.supports?(` for example https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/network_routers_controller.rb#L7

Reply:
@kavyanekkalapu https://github.com/ManageIQ/manageiq/pull/21181 adds the supports feature for you to use here

Reply:
@agrare thanks i updated it by looking at safe_delete func, please check and let me know if that is not sufficient.